mlx/dist/dist.h

```diff
@@ -0,0 +1,40 @@
+// Copyright © 2024 Apple Inc.
+
+#pragma once
+
+#include <memory>
+
+#include "mlx/array.h"
+
+namespace mlx::core::dist {
+
+/* Check if a communication backend is available */
+bool is_available();
+
+/* Return the communication stream. */
+Stream communication_stream();
```

Comment:
Maybe this should be an implementation detail for now rather than in the public API